tests/framework/crossplane.go

```diff
@@ -0,0 +1,218 @@
+	// don't care about, and we just want to check if a substring exists.
+	ValueSubstringAllowed bool
+}
+
+// ValidateNginxFieldExists accepts the nginx config and the configuration for the expected field,
+// and returns whether or not that field exists where it should.
+func ValidateNginxFieldExists(conf *Payload, expFieldCfg ExpectedNginxField) bool {
+	for _, config := range conf.Config {
+		if !strings.Contains(config.File, expFieldCfg.File) {
+			continue
+		}
+
+		for _, directive := range config.Parsed {
+			if len(expFieldCfg.Servers) == 0 {
+				if expFieldCfg.fieldFound(directive) {
+					return true
+				}
+				continue
+			}
+
+			for _, serverName := range expFieldCfg.Servers {
+				if directive.Directive == "server" && getServerName(directive.Block) == serverName {
+					for _, serverDirective := range directive.Block {
+						if expFieldCfg.Location == "" && expFieldCfg.fieldFound(serverDirective) {
+							return true
+						} else if serverDirective.Directive == "location" &&
+							fieldExistsInLocation(serverDirective, expFieldCfg) {
+							return true
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return false
+}
+
+func getServerName(serverBlock Directives) string {
+	for _, directive := range serverBlock {
+		if directive.Directive == "server_name" {
+			return directive.Args[0]
+		}
+	}
+
+	return ""
+}
+
+func (e ExpectedNginxField) fieldFound(directive *Directive) bool {
+	arg := strings.Join(directive.Args, " ")
+
+	valueMatch := arg == e.Value
+	if e.ValueSubstringAllowed {
+		valueMatch = strings.Contains(arg, e.Value)
+	}
+
+	return directive.Directive == e.Key && valueMatch
+}
+
+func fieldExistsInLocation(serverDirective *Directive, expFieldCfg ExpectedNginxField) bool {
+	// location could start with '=', so get the last element which is the path
+	loc := serverDirective.Args[len(serverDirective.Args)-1]
+	if loc == expFieldCfg.Location {
+		for _, locDirective := range serverDirective.Block {
+			if expFieldCfg.fieldFound(locDirective) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// injectCrossplaneContainer adds an ephemeral container that contains crossplane for parsing
+// nginx config. It attaches to the nginx container and shares volumes with it.
+func injectCrossplaneContainer(
+	k8sClient kubernetes.Interface,
+	timeout time.Duration,
+	ngfPodName,
+	namespace string,
+) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	pod := &core.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ngfPodName,
+			Namespace: namespace,
+		},
+		Spec: core.PodSpec{
+			EphemeralContainers: []core.EphemeralContainer{
+				{
+					TargetContainerName: "nginx",
+					EphemeralContainerCommon: core.EphemeralContainerCommon{
+						Name:            "crossplane",
+						Image:           "nginx-crossplane:latest",
+						ImagePullPolicy: "Never",
+						Stdin:           true,
+						VolumeMounts: []core.VolumeMount{
```

Comment:
We will have to update this list if we add a new config volume. Do you think it would help to add a reminder about that [here](https://github.com/nginxinc/nginx-gateway-fabric/blob/bf17bd50009b3eecc96162e302015ee894f1fb12/internal/mode/static/nginx/config/generator.go#L52)?